lib/ansible/modules/network/fortios/fortios_dlp_settings.py

```diff
@@ -0,0 +1,271 @@
+      password: "{{ password }}"
+      vdom:  "{{ vdom }}"
+      dlp_settings:
+        cache-mem-percent: "3"
+        chunk-size: "4"
+        db-mode: "stop-adding"
+        size: "6"
+        storage-device: "<your_own_value> (source system.storage.name)"
+'''
+
+RETURN = '''
+build:
+  description: Build number of the fortigate image
+  returned: always
+  type: str
+  sample: '1547'
+http_method:
+  description: Last method used to provision the content into FortiGate
+  returned: always
+  type: str
+  sample: 'PUT'
+http_status:
+  description: Last result given by FortiGate on last operation applied
+  returned: always
+  type: str
+  sample: "200"
+mkey:
+  description: Master key (id) used in the last call to FortiGate
+  returned: success
+  type: str
+  sample: "id"
+name:
+  description: Name of the table used to fulfill the request
+  returned: always
+  type: str
+  sample: "urlfilter"
+path:
+  description: Path of the table used to fulfill the request
+  returned: always
+  type: str
+  sample: "webfilter"
+revision:
+  description: Internal revision number
+  returned: always
+  type: str
+  sample: "17.0.2.10658"
+serial:
+  description: Serial number of the unit
+  returned: always
+  type: str
+  sample: "FGVMEVYYQT3AB5352"
+status:
+  description: Indication of the operation's result
+  returned: always
+  type: str
+  sample: "success"
+vdom:
+  description: Virtual domain used
+  returned: always
+  type: str
+  sample: "root"
+version:
+  description: Version of the FortiGate
+  returned: always
+  type: str
+  sample: "v5.6.3"
+
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+fos = None
+
+
+def login(data):
+    host = data['host']
+    username = data['username']
+    password = data['password']
+
+    fos.debug('on')
+    if 'https' in data and not data['https']:
+        fos.https('off')
+    else:
+        fos.https('on')
+
+    fos.login(host, username, password)
+
+
+def filter_dlp_settings_data(json):
+    option_list = ['cache-mem-percent', 'chunk-size', 'db-mode',
+                   'size', 'storage-device']
+    dictionary = {}
+
+    for attribute in option_list:
+        if attribute in json and json[attribute] is not None:
+            dictionary[attribute] = json[attribute]
+
+    return dictionary
+
+
+def dlp_settings(data, fos):
+    vdom = data['vdom']
+    dlp_settings_data = data['dlp_settings']
+    filtered_data = filter_dlp_settings_data(dlp_settings_data)
+    return fos.set('dlp',
+                   'settings',
+                   data=filtered_data,
+                   vdom=vdom)
+
+
+def fortios_dlp(data, fos):
+    login(data)
+
+    methodlist = ['dlp_settings']
+    for method in methodlist:
+        if data[method]:
+            resp = eval(method)(data, fos)
+            break
+
+    fos.logout()
+    return not resp['status'] == "success", resp['status'] == "success", resp
+
+
+def main():
+    fields = {
+        "host": {"required": True, "type": "str"},
```

Comment:
Create a common argument spec like [this](https://github.com/ansible/ansible/blob/d252cc7a2bd05fad2c694727c715df222b3ed58f/lib/ansible/module_utils/vmware.py#L450), other modules can reuse it.